modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -0,0 +1,699 @@
+
+#include <ade/util/zip_range.hpp>
+
+#include <openvino/openvino.hpp>
+
+#include <fstream>
+
+static ov::Core getCore() {
+    static ov::Core core;
+    return core;
+}
+
+static ov::element::Type toOV(int depth) {
+    switch (depth) {
+        case CV_8U:  return ov::element::u8;
+        case CV_32S: return ov::element::i32;
+        case CV_32F: return ov::element::f32;
+        case CV_16F: return ov::element::f16;
+        default: GAPI_Error("OV. Unsupported data type");
+    }
+    return ov::element::undefined;
+}
+
+static std::vector<int> toCV(const ov::Shape &shape) {
+    std::vector<int> result;
+    result.reserve(shape.size());
+    for (auto dim : shape) {
+        result.push_back(ade::util::checked_cast<int>(dim));
+    }
+    return result;
+}
+
+static int toCV(const ov::element::Type &type) {
+    switch (type) {
+        case ov::element::u8:  return CV_8U;
+        case ov::element::f32: return CV_32F;
+        case ov::element::i32: return CV_32S;
+        case ov::element::i64: return CV_32S;
+        case ov::element::f16: return CV_16F;
+        default: GAPI_Error("OV. Unsupported data type");
+    }
+    return -1;
+}
+
+static void copyFromOV(const ov::Tensor &tensor, cv::Mat &mat) {
+    if (tensor.get_byte_size() != mat.total() * mat.elemSize()) {
+        cv::util::throw_error(std::logic_error
+            ("OVBackend: Failed to copy data from ov::Tensor to cv::Mat! "
+             "Number of bytes mismatch. "
+             "(cv::Mat " + std::to_string(mat.total() * mat.elemSize()) + ", "
+             "ov::Tensor: " + std::to_string(tensor.get_byte_size()) + ")"));
+    }
+    std::copy_n(reinterpret_cast<uint8_t*>(tensor.data()),
+                tensor.get_byte_size(),
+                mat.ptr<uint8_t>());
+}
+
+static void copyToOV(const cv::Mat &mat, ov::Tensor &tensor) {
+    if (tensor.get_byte_size() != mat.total() * mat.elemSize()) {
+        cv::util::throw_error(std::logic_error
+            ("OVBackend: Failed to copy data from cv::Mat to ov::Tensor! "
+             "Number of bytes mismatch. "
+             "(cv::Mat " + std::to_string(mat.total() * mat.elemSize()) + ", "
+             "ov::Tensor: " + std::to_string(tensor.get_byte_size()) + ")"));
+    }
+    std::copy_n(mat.ptr<uint8_t>(),
+                tensor.get_byte_size(),
+                reinterpret_cast<uint8_t*>(tensor.data()));
+}
+
+std::vector<int> cv::gapi::ov::util::to_ocv(const ::ov::Shape &shape) {
+    return toCV(shape);
+}
+
+int cv::gapi::ov::util::to_ocv(const ::ov::element::Type &type) {
+    return toCV(type);
+}
+
+//struct IUnit {
+    //virtual std::string get_any_input_name() = 0;
+    //virtual std::string get_any_output_name() = 0;
+//}
+
+using ParamDesc = cv::gapi::ov::detail::ParamDesc;
+struct OVUnit {
+    static const char *name() { return "OVUnit"; }
+
+    explicit OVUnit(const ParamDesc &pd)
+        : params(pd) {
+
+        if (cv::util::holds_alternative<ParamDesc::IR>(params.kind)) {
```

Comment:
Probably we need this logic encapsulated somewhere not to check it everywhere!!!